docfx/doxygen2markdown.cc

```diff
@@ -0,0 +1,60 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "docfx/doxygen2markdown.h"
+#include <iostream>
+#include <sstream>
+#include <string_view>
+
+namespace {
+
+[[noreturn]] void UnknownChildType(std::string_view where,
+                                   pugi::xml_node const& child) {
+  std::ostringstream os;
+  os << "Unknown child in " << where << "(): name=" << child.name();
+  os << ", attributes=[";
+  for (auto const& a : child.attributes()) {
+    os << ", " << a.name() << "=" << a.as_string();
```

Comment:
Avoid separator on first attribute?

Reply:
Good idea.  I am just going to rely on `child.print(os);` to do all the work (which it does).